core/platform/auth/firebase_auth_services.py

```diff
@@ -136,6 +128,40 @@ def authenticate_request(request):
     return None if not auth_id else auth_domain.AuthClaims(auth_id, email)
 
 
+def delete_associations(user_id):
+    """Deletes associations referring to the given user_id.
+
+    Args:
+        user_id: str. The ID of the user being deleted.
+
+    Returns:
+        bool. Whether the user's data has been successfully deleted.
+    """
+    assoc = auth_models.UserIdByFirebaseAuthIdModel.get_by_user_id(user_id)
+    if assoc is None:
+        return
+    try:
+        with _acquire_firebase_context():
+            firebase_auth.delete_user(assoc.id)
+    except (ValueError, firebase_exceptions.FirebaseError) as e:
+        logging.exception(e)
+
+
+def are_associations_deleted(user_id):
```

Comment:
I'm a bit confused about the singular/plural here. Is there one association or multiple associations? The function names suggest the latter but the implementation logic suggests the former.

Perhaps rename 'assoc' to 'associations' / 'association' to make it clearer whether we are talking about singular/plural here.